src/OpenSage.Game/Gui/Apt/SpriteItem.cs

```diff
@@ -65,24 +67,45 @@ public override void Create(Character character, AptContext context, SpriteItem
                             FrameLabels[fl.Name] = fl.FrameId;
                             break;
                     }
-
                 }
             }
         }
 
-        public override void Render(AptRenderer renderer, ItemTransform pTransform, DrawingContext2D dc)
+        protected override void RenderImpl(AptRenderingContext renderingContext)
         {
             if (!Visible)
                 return;
 
             //calculate the transform for this element
-            var cTransform = pTransform * Transform;
+            renderingContext.PushTransform(Transform);
+
+            var clipMask = (Texture) null;
+            var clipDepth = 0;
 
             //render all subItems
-            foreach (var item in Content.Items.Values)
+            foreach (var (depth, item) in Content.Items)
             {
-                item.Render(renderer, cTransform, dc);
+                item.Render(renderingContext);
+
+                if (depth > clipDepth && clipMask != null)
+                {
+                    renderingContext.SetClipMask(null);
+                    clipDepth = 0;
+                }
+
+                if (item.ClipDepth != null)
+                {
+                    renderingContext.SetClipMask(item.ClipMask);
+                    clipDepth = item.ClipDepth.Value;
+                }
+            }
+
+            // In case the clipMask wans't cleared inside the loop
+            if (clipDepth > 0)
```

Comment:
@tgjones you were missing that line that's why the clipmask was used by parent displayitems